spec/awful/completion_spec.lua

```diff
@@ -60,4 +60,22 @@ describe("awful.completion.shell", function()
     end)
 end)
 
+describe("awful.completion.shell without $SHELL", function()
+    local orig_popen = io.popen
+
+    setup(function()
+        io.popen = function(...)  --luacheck: ignore
+            return orig_popen('env -u SHELL ' .. ...)
```

Comment:
Uhm... this unsets `$SHELL` inside of the shell. This does not have any effects inside of awesome, no?

You could monkey-patch `os.getenv` instead, but doesn't `awful.completion` cache the result of looking at `os.getenv`?